projects/plugins/jetpack/_inc/client/components/popover/index.jsx

```diff
@@ -76,12 +80,18 @@ class Popover extends Component {
 		bindWindowListeners();
 	}
 
+	isDOMNode( obj ) {
+		return obj instanceof HTMLElement;
+	}
+
 	UNSAFE_componentWillReceiveProps( nextProps ) {
-		// update context (target) reference into a property
-		if ( ! isDOMElement( nextProps.context ) ) {
-			this.domContext = ReactDom.findDOMNode( nextProps.context );
-		} else {
-			this.domContext = nextProps.context;
+		// update context (target - expecting a DOM node not a ref) reference into a property
+
+		if ( this.isDOMNode( nextProps.context ) ) {
+			this.domContextRef.current = nextProps.context;
+		} else if ( nextProps.context !== null ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Expected a DOM node for props.context', nextProps.context );
 		}
```

Comment:
Looking at this again, we lost assigning the null if it is null. I think I left that out of my original suggestion.

Now that we dropped handling refs, it's easy to fix.

```suggestion
		if ( nextProps.context === null || this.isDOMNode( nextProps.context ) ) {
			this.domContextRef.current = nextProps.context;
		} else {
			// eslint-disable-next-line no-console
			console.error( 'Expected a DOM node for props.context', nextProps.context );
		}
```

Reply:
Or, as @sirreal mentioned in a comment while I was writing this, maybe we should just check for `! nextProps.context` instead of `nextProps.context === null` to handle undefined too.

Reply:
Nice catch, thank you.